pkg/validator/patch.go

```diff
@@ -130,7 +131,7 @@ func createOrUpdateValidatingWebhook(clientSet kubernetes.Interface, cert *certi
 						Port:      &webhookPort,
 					},
 					CABundle: cert.GetTrustedCAs()},
-				FailurePolicy: &failurePolicy,
+				FailurePolicy: &ignoreConnectionFailurePolicy,
```

Comment:
Why this change and why here?

Reply:
If enableMeshRootCertificate is set to true on OSM install an MRC will be created by the osm-bootstrap. The osm-controller and osm-injector will not have started up since they are waiting on the osm-bootstrap to be ready. If the validating webhook's failure policy is fail on failed connection to validating webhook server, MRC creation will fail in the osm-bootstrap since the osm-controller hasn't started up and therefore the validating webhook server hasn't been created. By ignoring connection failures specifically for the validation of the MRC, the osm-bootstrap can successfully create an MRC regardless of whether or not the osm-controller is ready.

Reply:
This is part of the challenge of the way these are handled in the bootstrap.  How is this done for the meshconfiguraiton crd?

The issue with the ignore failure policy means that if for some reason the webhook is down later on down the road then we could allow multiple active MRCs into the system.

Reply:
We currently don't validate the MeshConfig.

Reply:
Is there a reason to use the bootstrapper to deploy the initial MRC?  

Could we include the yaml for MRC instead? I think it is pretty static?

Reply:
> Is there a reason to use the bootstrapper to deploy the initial MRC?
> 
> Could we include the yaml for MRC instead? I think it is pretty static?

Since we don't apply the CRDs using helm I can't create any CRs before applying the CRDs in the osm-bootstrap.

Reply:
As long as the ValidatingWebhookConfiguration has not been applied with the control plane validating webhook or the osm-controller is running, MRC creation will succeed. Since the osm-controller is responsible for creating the ValidatingWebhookConfiguration and the osm-controller hasn't started up, the MRC can be created by the osm-bootstrap during installation.

Reply:
I think this is less of an issue than I thought it was. I was concerned about the fresh install scenario, but I think this scenario is most applicable to enabling the `enableMeshRootCertificate` feature flag after install, and restarting all control plane components. 

Speaking with @jsturtevant, one possible solution to avoiding this timing issue is to only add the control plane webhook if `enableMeshRootCertificate` is set to true.

Reply:
Looks like this was dropped and we are still in good shape? The restart scenario was passing?  We also have tests covering the other scenario, where we do a fresh install and those are passing.

Reply:
Yep, I reverted the policy back to FAIL. The e2e that does a rolling restart is passing. I think conditionally applying the control plane webhook would be a nice feature to have, but I don't think the additional complexity is worth it in this case.